tests/test_routing.py

```diff
@@ -40,6 +40,28 @@ def test_pit_filling(self):
         dem_array[3:8, 3:8] = 0.0
         numpy.testing.assert_almost_equal(result_array, dem_array)
 
+    def test_pit_filling_small_delta(self):
+        """PGP.routing: test pitfilling on small delta."""
+        base_path = os.path.join(self.workspace_dir, 'base.tif')
+        dem_array = numpy.empty((4, 4), dtype=numpy.float32)
+        # these values came from a real world dem that failed
+        lower_val = 272.53228759765625
+        higher_val = 272.5325012207031
+        dem_array[:] = higher_val
+        dem_array[2, 2] = lower_val
+
+        expected_result = numpy.empty((4, 4), numpy.float32)
+        expected_result[:] = higher_val
+        _array_to_raster(dem_array, None, base_path)
+        fill_path = os.path.join(self.workspace_dir, 'filled.tif')
+        pygeoprocessing.routing.fill_pits(
+            (base_path, 1), fill_path, working_dir=self.workspace_dir)
+        result_array = pygeoprocessing.raster_to_numpy_array(fill_path)
+        numpy.set_printoptions(suppress=True)
+        self.assertTrue(
+            (result_array == expected_result).all(),
+            result_array == expected_result)
+
```

Comment:
This looked a little strange at first with two args inside `assertTrue`. Then I tried it out and see how it makes for a very useful failure report!

Reply:
ha glad you like it! as you might guess from your previous comment i'd tried a few things.